libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryService.java

```diff
@@ -287,8 +290,26 @@ private boolean locationPermissionCheck() {
       int finePermission = PermissionChecker.checkSelfPermission(MapboxTelemetry.applicationContext,
         Manifest.permission.ACCESS_FINE_LOCATION);
 
+      if (checkFinePermission(finePermission)) {
+        return false;
+      }
+
       return coarsePermission == PackageManager.PERMISSION_GRANTED
         || finePermission == PackageManager.PERMISSION_GRANTED;
     }
   }
+
+  private boolean checkFinePermission(int finePermission) {
+    if (finePermission != PackageManager.PERMISSION_GRANTED) {
+      try {
+        throw new Exception(MISSING_FINE_PERMISSION);
```

Comment:
i'm still not sure why we need to throw exception catch and handle it in the same method? since we're basically stuck with passive provider would it make sense just to check if fine permissions are granted and if not let the user know that we can't collect location preferably via SecurityException? Thoughts?